### ISSUE TITLE
Connect-DbaInstance - Returned object is missing properties if -SqlInstance is of type Microsoft.SqlServer.Management.Smo.Server

### ISSUE THREAD

**Issue by andreasjordan** [sqlcollaborative/dbatools#6811]
If -SqlInstance is a string or of type DbaInstanceParameter, then everything is ok.

But if I accidentaly use a plain [Microsoft.SqlServer.Management.Smo.Server] object that has not yet the dbatools specific properties like ComputerName, they are NOT added by Connect-DbaInstance.

Is this a bug or a feature?
I would expect that Connect-DbaInstance always returns full objects with dbatools specific properties.

These steps should work on any system, but I ran them on en-US windows server 2016:

```
Import-Module -Name dbatools -MinimumVersion 1.0.116

# This is the official way, that works as expected:
$instObjectDba = [DbaInstanceParameter]'SRV1\SQL2017'
$instConnectDba = Connect-DbaInstance -SqlInstance $instObjectDba
$instObjectDba | Format-Table -Property *Name*
$instConnectDba | Format-Table -Property *Name*
$instConnectDba.Query('SELECT @@SERVERNAME AS Servername').Servername

<# Output:

ComputerName InstanceName FullName     FullSmoName  SqlComputerName SqlInstanceName SqlFullName   
------------ ------------ --------     -----------  --------------- --------------- -----------   
SRV1         SQL2017      SRV1\SQL2017 SRV1\SQL2017 [SRV1]          [SQL2017]       [SRV1\SQL2017]



ComputerName DbaInstanceName ClusterName ComputerNamePhysicalNetBIOS FilestreamShareName InstanceName NamedPipesEnabled NetName ServiceName SqlCharSetName SqlSortOrderName Name         DomainName DomainInstanceName
------------ --------------- ----------- --------------------------- ------------------- ------------ ----------------- ------- ----------- -------------- ---------------- ----         ---------- ------------------
SRV1         SQL2017         SQLCluster  SRV1                        SQL2017             SQL2017                  False SRV1    SQL2017     iso_1          nocase_iso       SRV1\SQL2017 SMO        SRV1\SQL2017      


SRV1\SQL2017

#>

# What if I accidentaly use a plain [Microsoft.SqlServer.Management.Smo.Server]? 
$instObjectSmo = [Microsoft.SqlServer.Management.Smo.Server]'SRV1\SQL2017'
$instConnectSmo = Connect-DbaInstance -SqlInstance $instObjectSmo
$instObjectSmo | Format-Table -Property *Name*
$instConnectSmo | Format-Table -Property *Name*
$instConnectSmo.Query('SELECT @@SERVERNAME AS Servername').Servername

<# Output:

ClusterName ComputerNamePhysicalNetBIOS FilestreamShareName InstanceName NamedPipesEnabled NetName ServiceName SqlCharSetName SqlSortOrderName Name         DomainName DomainInstanceName
----------- --------------------------- ------------------- ------------ ----------------- ------- ----------- -------------- ---------------- ----         ---------- ------------------
SQLCluster  SRV1                        SQL2017             SQL2017                  False SRV1    SQL2017     iso_1          nocase_iso       SRV1\SQL2017 SMO        SRV1\SQL2017      



ClusterName ComputerNamePhysicalNetBIOS FilestreamShareName InstanceName NamedPipesEnabled NetName ServiceName SqlCharSetName SqlSortOrderName Name         DomainName DomainInstanceName
----------- --------------------------- ------------------- ------------ ----------------- ------- ----------- -------------- ---------------- ----         ---------- ------------------
SQLCluster  SRV1                        SQL2017             SQL2017                  False SRV1    SQL2017     iso_1          nocase_iso       SRV1\SQL2017 SMO        SRV1\SQL2017      


SRV1\SQL2017

#>

# I can use the object, but I miss dbatools specific properties like ComputerName.
```

**Comment by wsmelton** [comment]
This is by design. The purpose of the connect command is to output the SMO server object. It will accept input of string or object only. The computer property does not get added in this function.

**Comment by potatoqualitee** [comment]
i'd say this is both by design and a bug 😅  it just checks if it's an SMO object and returns it, expecting that the object was created by dbatools. I didn't consider someone creating a simple class and passing it on. Connect-DbaInstance could use a fresh set of eyes, please feel free to check for our standard properties and add them if they are missing. Perhaps even extracting the connectionstring from $server.ConnectionProperties (I forget the property) then just repassing it to Connect-DbaInstance.

**Comment by wsmelton** [comment]
#6246

**Comment by andreasjordan** [comment]
Solved with the experimental code path from #6904.
